app/controllers/api/v2/roles_controller.rb

```diff
@@ -51,11 +51,30 @@ def destroy
         process_response @role.destroy
       end
 
+      api :POST, "/roles/:id/clone", N_("Clone a role")
+      param :id, String, :required => true
+      param :new_name, String, :required => true, :desc => N_('Name of the cloned role')
```

Comment:
I think all attributes should be consistent. Description is cloned and would not be appended if I specify new one correct/ So I think orgs and locs should behave the same. If you don't specify any, it should have the same as origin. If I specify some (including empty array), it should set this new collection.